src/codewhisperer/util/authUtil.ts

```diff
@@ -138,22 +139,48 @@ export class AuthUtil {
         if (this.isConnectionExpired()) {
             try {
                 await this.auth.reauthenticate(this.conn!)
+                this.refreshCodeWhisperer()
             } catch (err) {
                 throw ToolkitError.chain(err, 'Unable to authenticate connection')
             }
         }
     }
 
-    public async showReauthenticatePrompt() {
+    public async refreshCodeWhisperer() {
+        await Promise.all([
+            vscode.commands.executeCommand('aws.codeWhisperer.refresh'),
+            vscode.commands.executeCommand('aws.codeWhisperer.refreshRootNode'),
+            vscode.commands.executeCommand('aws.codeWhisperer.refreshStatusBar'),
```

Comment:
any reason  `aws.codeWhisperer.refresh` couldn't also call aws.codeWhisperer.refreshRootNode and aws.codeWhisperer.refreshStatusBar ?